packages/core/src/components/product/ProductGrid/ProductGrid.tsx

```diff
@@ -26,45 +27,105 @@ interface Props {
     ProductCardProps,
     'showDiscountBadge' | 'bordered' | 'taxesConfiguration'
   >
+  /**
+   * Identify the number of firstPage
+   */
+  firstPage?: number
 }
 
 function ProductGrid({
   products,
   page,
   pageSize,
   productCard: { showDiscountBadge, bordered, taxesConfiguration } = {},
+  firstPage,
 }: Props) {
   const { __experimentalProductCard: ProductCard } =
     useOverrideComponents<'ProductGallery'>()
   const aspectRatio = 1
 
+  // TODO: Check if is also isMobile
+  const isFirstPage = firstPage === page
```

Comment:
We can remove this improvement on this PR, and adds its lately. 
We should use the `useScreenResize` hook from https://github.com/vtex/faststore/pull/2487